### PR TITLE
Build boat in allied shipyards is allowed now

### PR DESCRIPTION
**Pull request by nullkiller**
One more freeze. AI wanted to build boat in allied shipyard. It was not allowed on server

### CODE REVIEW

File: server/NetPacksServer.cpp [hunk 1]
```diff
@@ -272,7 +272,9 @@ bool HireHero::applyGh(CGameHandler * gh)
 
 bool BuildBoat::applyGh(CGameHandler * gh)
 {
-	throwOnWrongOwner(gh, objid);
+	if(gh->getPlayerRelations(gh->getOwner(objid), gh->getPlayerAt(c)) == PlayerRelations::ENEMIES)
+		throwAndCompain(gh, "Can't build boat at enemy shipyard");
```
Comment (DjWarmonger) [inline server/NetPacksServer.cpp]: Isn't this a typo?

Comment (nullkiller) [reply server/NetPacksServer.cpp]: Sure it is. Wanna me to fix it?

Comment (nullkiller) [reply server/NetPacksServer.cpp]: I also think I can extract a method throwWnenOwnerIsEnemy and use it for recruit hero

Comment (nullkiller) [reply server/NetPacksServer.cpp]: I also think I can extract a method throwWnenOwnerIsEnemy and use it for recruit hero

**Pull request merged**